File: src/Http.API/ClientApp/services/auth.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { LoginDto } from '../models/auth/login-dto.model';
import { AuthResult } from '../models/auth/auth-result.model';

/**
 * Auth
 */
@Injectable({ providedIn: 'root' })
export class AuthService extends BaseService {
  /**
   * login
   * @param data LoginDto
   */
  login(data: LoginDto): Observable<AuthResult> {
    const url = `/api/Auth`;
    return this.request<AuthResult>('post', url, data);
  }

  /**
   * logout
   * @param id string
   */
  logout(id: string): Observable<boolean> {
    const url = `/api/Auth/${id}`;
    return this.request<boolean>('get', url);
  }

}

File: src/Sdks/Dusi.Manage.Client/Services/Authservice.cs
using Share.Models.AuthDtos;
namespace Dusi.Manage.Client.Services;
/// <summary>
/// 系统用户授权登录
/// </summary>
public class AuthService : BaseService
{
    public AuthService(HttpClient httpClient) : base(httpClient)
    {

    }
    /// <summary>
    /// 登录获取Token
    /// </summary>
    /// <param name="data">LoginDto</param>
    /// <returns></returns>
    public async Task<AuthResult?> LoginAsync(LoginDto data)
    {
        string url = $"/api/admin/Auth";
        return await PostJsonAsync<AuthResult?>(url, data);
    }

    /// <summary>
    /// 退出
    /// </summary>
    /// <param name="id">string </param>
    /// <returns></returns>
    public async Task<bool?> LogoutAsync(string id)
    {
        string url = $"/api/admin/Auth/{id}";
        return await GetJsonAsync<bool?>(url);
    }

}